include/pre-charge/pre_charge.hpp

```diff
@@ -0,0 +1,202 @@
+        // When the MC is discharging
+        DISCHARGE = 4u,
+        // When the contactor is open
+        CONT_OPEN = 5u,
+        // When the contactor is closed
+        CONT_CLOSE = 6u,
+        // When the forward enable is disabled
+        FORWARD_DISABLE = 7u
+    };
+
+    /**
+     * Constructor for pre-charge state machine
+     * 
+     */
+    pre_charge(IO::GPIO& key, IO::GPIO& batteryOne, IO::GPIO& batteryTwo,
+               IO::GPIO& eStop, IO::GPIO& pc, IO::GPIO& dc, IO::GPIO& cont,
+               IO::GPIO& apm, IO::GPIO& forward);
+
+    /**
+     * Handler running the pre-charge state switching
+     * 
+     */
+    void handle();
+
+    /**
+    * Get the value of STO (Safe to Operate)
+    * 
+    * STO=1 when BATTERY_1_OK=1 AND BATTERY_2_OK=1 AND EMERGENCY_STOP=1
+    * 
+    * @return value of STO, 0 or 1
+    */
+    void getSTO();
+
+    /**
+     * Get the state of MC_KEY_IN
+     * 
+     */
+    void getMCKey();
+
+    /**
+     * Set the Precharge state
+     * 
+     * @param state 0 = precharge disabled, 1 = precharge enabled
+     */
+    void setPrecharge(int state);
+
+    /**
+     * Set the Discharge state
+     * 
+     * @param state 0 = discharge disabled, 1 = discharge enabled
+     */
+    void setDischarge(int state);
+
+    /**
+     * Toggle the state of the Main Contactor
+     * 
+     * @param state 0 = contactor open, 1 = contactor closed
+     */
+    void setContactor(int state);
+
+    /**
+     * Set the Forward state
+     * 
+     * @param state 0 = forward disabled, 1 = forward enabled
+     */
+    void setForward(int state);
+
+    /**
+     * Set the APM state
+     * 
+     * @param state 0 = APM disabled, 1 = APM enabled
+     */
+    void setAPM(int state);
+
+    /**
+     * Handles when the MC is powered off.
+     * 
+     * State: State::MC_OFF
+     */
+    void mcOffState();
+
+    /**
+     * Handles when the MC is powered on.
+     * 
+     * State: State::MC_ON
+     */
+    void mcOnState();
+
+    /**
+     * Handles when an unsafe condition is detected while the MC is off.
+     * 
+     * State: State::ESTOPWAIT
+     */
+    void eStopState();
+
+    /**
+     * Handles when precharge is to occur.
+     * 
+     * State: State::PRECHARGE
+     */
+    void prechargeState();
+
+    /**
+     * Handles when discharge is to occur.
+     * 
+     * State: State::DISCHARGE
+     */
+    void dischargeState();
+
+    /**
+     * Handles when the contactor is to be open.
+     * 
+     * State: State::CONT_OPEN
+     */
+    void contOpenState();
+
+    /**
+     * Handles when the contactor is to be closed.
+     * 
+     * State: State::CONT_CLOSE
+     */
+    void contCloseState();
+
+    /**
+     * Handles when the Forward Enable is to be disabled.
+     * 
+     * State: State::FORWARD_DISABLE
+     */
+    void forwardDisableState();
+
+    /**
+     * Pretty print the current state machine status for testing
+     * 
+     */
+    std::string printState();
```

Comment:
Cannot use `std::string` since we don't have dynamic memory. Instead the function should take in a character array to populate along with a max size